File: extract_js.py
import sys, re
with open('public/modal_ai_studio.html', 'r', encoding='utf-8') as f:
    content = f.read()

scripts = re.findall(r'<script>(.*?)</script>', content, re.DOTALL)
for i, s in enumerate(scripts):
    with open(f'script_{i}.js', 'w', encoding='utf-8') as out:
        out.write(s)

File: extract_js.js
const fs = require('fs');
const html = fs.readFileSync('frontend/modal_ai_studio.html', 'utf-8');
const scriptRegex = /<script\b[^>]*>([\s\S]*?)<\/script>/gm;
let match;
let i = 0;
while ((match = scriptRegex.exec(html)) !== null) {
    const code = match[1];
    fs.writeFileSync('temp_script_' + i + '.js', code);
    console.log('Extracted script ' + i);
    i++;
}
